docs/v3/core/v3.0.rst

```diff
@@ -1211,40 +1210,42 @@ the following procedure:
 
 4. The chunk array ``A`` is equal to ``EC[0]``.
 
+.. _codec-specification:
+
 Specifying codecs
 -----------------
 
-To allow for flexibility to define and implement new codecs, this
-specification does not define any codecs, nor restrict the set of
-codecs that may be used. Each codec must be defined via a separate
-specification. In order to refer to codecs in array metadata
-documents, each codec must have a unique identifier, which is a URI
-that dereferences to a human-readable specification of the codec. A
-codec specification must declare the codec identifier, and describe
-(or cite documents that describe) the encoding and decoding algorithms
-and the format of the encoded data.
-
-A codec may have configuration parameters which modify the behaviour
-of the codec in some way. For example, a compression codec may have a
-compression level parameter, which is an integer that affects the
-resulting compression ratio of the data. Configuration parameters must
-be declared in the codec specification, including a definition of how
-configuration parameters are represented as JSON.
-
-The Zarr core development team maintains a repository of codec
-specifications, which are hosted alongside this specification in the
-`zarr-specs GitHub repository`_, and which are
-published on the `zarr-specs documentation Web site
-<https://zarr-specs.readthedocs.io/>`_. For ease of discovery, it is
-recommended that codec specifications are contributed to the
-zarr-specs GitHub repository. However, codec specifications may be
-maintained by any group or organisation and published in any location
-on the Web. For further details of the process for contributing a
-codec specification to the zarr-specs GitHub repository, see
-`ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ which describes
-the process for Zarr specification changes.
-
-Further details of how codecs are configured for an array are given in the `Array metadata`_ section.
+.. _core-codecs:
+
+Core codecs
+-----------
+
+This spec defines a set of codecs ("core codecs") which all Zarr implementations SHOULD implement in 
+order to ensure a minimal level of interoperability between Zarr implementations.
+Each core codec is each defined by specification documents which are hosted in the
+`zarr-specs GitHub repository`_, and are published on the `zarr-specs documentation web site
+<https://zarr-specs.readthedocs.io/>`_. The list of core codecs is part of the Zarr v3 specification. 
+Changes to the list of core codecs MUST be made via the same protocol used for 
+changing the Zarr v3 specification. Changes to the list of core codecs SHOULD be made carefully and
+in close collaboration with extant Zarr v3 implementations. A new core codec SHOULD be added to the 
+list when a sufficient number of Zarr implementations support or intend to support that codec. 
```

Comment:
What is sufficient? 50%? 80%? 100%? This should be clarified.

Reply:
Any number we pick will be arbitrary. I think we have to rely on people's judgment here.

Reply:
I think picking and agreeing on an arbitrary number is better than not picking a number and then leaving the door open to discussions/disagreements in the future on what a "sufficient number" means.

Reply:
I guess because it's a "SHOULD" we can put in a completely arbitrary number and then ignore it. How about 60%

Reply:
or "an overwhelming majority"?

Reply:
Maybe it's worth removing this sentence completely? I don't particularly like adding an arbitrary number either as per your comments 😄 . I think it would be fine to remove this sentence, and if someone wants to add a core codec they can justify it as part of the normal ZEP process.